### DISCUSSION TITLE
Add a control that will encapsulate HTML and all reated Styles, Scripsts, etc.

### DISCUSSION THREAD

**Discussion by yogeshvachhani** [frappe/builder#275] [Ideas]
Lately I have been using WordPress a lot for building CMS based web sites.

In WordPress it is very hard to add highly interactive controls (widgets) in a page or post and if we try to do this it requires lots of tweaking and coding. But to make developers life easier WordPress has provided a widget called _**Custom HTML**_ which allow one to enter a valid HTML document and it will get rendered on the page as the place where the widget is placed.

Here is a screen shots of the Custom HTML widget with content in HTML...
![image](https://github.com/user-attachments/assets/cdc74566-80ed-4680-9a83-6fb489a88ac0)

When the HTML is rendered it looks like this...
![image](https://github.com/user-attachments/assets/3d5d24b9-1662-4f18-8308-9785d3c43412)

Personally I feel adding such a widget will remove the burden of providing lots of various widgets/controls and interactions from your shoulders and it will give users the freedom to embed any valid HTML document withing a page.

Consider _Custom HTML_ like an iFrame that most of us are used to using. But in this case the content of the iFrame is embedded with itself instead of it coming from an external link.

I hope I am making sense.

If you need any for clarifications please let me know and I will be happy to share the same.

**Comment by batonac** [comment]
This already exists. Add an "Embed" block to your page, then double click to embed any kind of HTML code, including JS and CSS in corresponding tags.
